### PR TITLE
add caffenet

### PR DESCRIPTION
**Pull request by jkamelin**

### CODE REVIEW

File: models/public/caffenet/caffenet.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# caffenet
+
+## Use Case and High-Level Description
+
+CaffeNet model. Used for classification. For deatails see [paper](https://arxiv.org/abs/1408.5093).
```
Comment (IRDonch) [inline models/public/caffenet/caffenet.md]: details*

File: tools/downloader/list_topologies.yml [hunk 2]
```diff
@@ -1570,6 +1570,31 @@ topologies:
...
+      CaffeNet model. Used for classification. For deatails see paper
+      <https://arxiv.org/abs/1408.5093>.
+    task_type: "classification"
+    files:
+      - name: bvlc_reference_caffenet.caffemodel
+        sha256: 472d4a06035497b180636d8a82667129960371375bd10fcb6df5c6c7631f25e0
+        source: http://dl.caffe.berkeleyvision.org/bvlc_reference_caffenet.caffemodel
+      - name: deploy.prototxt
+        sha256: 922248a4d2f6aac1cc8e7e5dbd996cc2ecd3356480d67c198f6cf96b12311a04
+        source: https://raw.githubusercontent.com/BVLC/caffe/master/models/bvlc_reference_caffenet/deploy.prototxt
```
Comment (IRDonch) [inline tools/downloader/list_topologies.yml]: Please use a specific commit rather than master.

**Comment by jkamelin** [comment]
cc @kchechil

**Comment by kchechil** [comment]
@jkamelin please resolve conversations, @IRDonch please merge if no more comments.

File: tools/downloader/list_topologies.yml [hunk 3]
```diff
@@ -1570,6 +1570,31 @@ topologies:
...
+      - --framework=caffe
+      - --data_type=FP32
+      - --input_shape=[1,3,227,227]
+      - --input=data
+      - --mean_values=data[104.0,117.0,123.0]
+      - --output=prob
+      - --input_model=$dl_dir/bvlc_reference_caffenet.caffemodel
+      - --input_proto=$dl_dir/deploy.prototxt
+    framework: "caffe"
+    license: https://raw.githubusercontent.com/BVLC/caffe/master/LICENSE
```
Comment (IRDonch) [inline tools/downloader/list_topologies.yml]: This isn't the correct license. The license for the model itself is in https://raw.githubusercontent.com/BVLC/caffe/master/models/bvlc_reference_caffenet/readme.md. See `alexnet` for a similar example.

File: tools/downloader/license.txt [hunk 4]
```diff
@@ -36,6 +36,38 @@ License terms:
...
+   This model is snapshot of iteration 310,000.
+   The best validation performance during training was iteration 313,000 with validation accuracy 57.412% and loss 1.82328.
+   This model obtains a top-1 accuracy 57.4% and a top-5 accuracy 80.4% on the validation set, using just the center crop.
+   (Using the average of 10 crops, (4 + 1 center) * 2 mirror, should obtain a bit higher accuracy still.)
+
+   This model was trained by Jeff Donahue @jeffdonahue
+
+   ## License
+
+   This model is released for unrestricted use.
```
Comment (IRDonch) [inline tools/downloader/license.txt]: This sentence is the only part that is the actual license. The rest of the file doesn't need to be here. Again, see `alexnet`.

**Pull request merged**